version: "3.9"

services:
  api:
    build: .
    command: >
      bash -c "python manage.py makemigrations && 
      python manage.py migrate && 
      python manage.py runserver 0.0.0.0:8000"
    # volumes:
    #  - .:/code
    ports:
      - 3888:8000
    depends_on:
      db:
        condition: service_healthy
    networks:
      - customnetwork
    restart: always
    container_name: sivalanet_django_server
  db:
    image: postgis/postgis:13-3.3 #postgres:13
    container_name: sivalanet_postgres_db

    healthcheck:
      test: "exit 0"
      interval: 1s
      timeout: 1s
      retries: 30

    platform: linux/amd64
    # restart: always
    volumes:
      - postgres_data:/var/lib/postgresql/data/
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
    ports:
      - 5455:5432
    environment:
      POSTGRES_PASSWORD: 'postgres'
      # - "POSTGRES_HOST_AUTH_METHOD=trust"
    networks:
      customnetwork:
        ipv4_address: 172.27.0.2
      
    restart: always
volumes:
  postgres_data:

networks:
  customnetwork:
    ipam:
      config:
        - subnet: 172.27.0.0/16